variables.tf

```
@@ -2,18 +2,25 @@ variable "public_ip_config" {
   type = object({
     enabled           = bool
     allocation_method = optional(string, "Static")
+    stage             = optional(string)
   })
   default = {
     enabled = false
   }
+  nullable = false
   validation {
     condition     = contains(["Static", "Dynamic"], var.public_ip_config.allocation_method)
     error_message = "Allocation method must be Static or Dynamic"
   }
+  validation {
+    condition     = var.public_ip_config.enabled == true ? var.public_ip_config.stage != null : true
+    error_message = "If public ip is enabled, stage must be set."
+  }
```

```suggestion
  type = object({
    allocation_method = optional(string, "Static")
    stage             = string
  })
  default = null
  nullable = false
  validation {
    condition     = var.public_ip_config == null ? true : contains(["Static", "Dynamic"], var.public_ip_config.allocation_method)
    error_message = "Allocation method must be Static or Dynamic"
  }
```

As we now got a required property, we can get rid of the annoying "enabled" attribute, by checking, if the whole object is null.

---

Speaking of which. We should probably also fix it in windows-vm. This one currently uses an incorrect naming convention: https://github.com/qbeyond/terraform-azurerm-windows-vm/blob/c046a22ff92cb9bb7ea002109985c2efda9e1dc6/locals.tf#L3

---

yes, please !!
Could you please, review and adapt my VM Windows PR https://github.com/qbeyond/terraform-azurerm-windows-vm/pull/14 , to have similar variables, parameters, etc,etc ?